test/api_getProduct_test.dart

```diff
@@ -763,5 +772,86 @@ void main() {
       expect(vegetableFat.vegetarian, IngredientSpecialPropertyStatus.POSITIVE);
       expect(vegetableFat.fromPalmOil, IngredientSpecialPropertyStatus.MAYBE);
     });
+
+    test('matched product', () async {
+      final Map<String, String> attributeImportances = {};
+      int refreshCounter = 0;
+      final ProductPreferencesManager manager = ProductPreferencesManager(
+        ProductPreferencesSelection(
+          (String attributeId, String importanceIndex) async {
```

Comment:
I'm thinking it would be good to use named parameters for ProductPreferencesSelection() so that this code can be understood when read, without having to know the parameters. (I already had forgotten what the 3 parameters were).

Reply:
Good point. I'll do that.